daemon/inertiad/notifier/notifier.go

```diff
@@ -0,0 +1,104 @@
+package notifier
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// Notifier manages notifications
+type Notifier interface {
+	Notify(string) error
+}
+
+// SlackNotifier represents slack notifications
+type SlackNotifier struct {
+	hookURL string
+}
+
+// NewNotifier creates a notifier with web hook url to slack channel
+func NewNotifier(webhook string) *SlackNotifier {
+	url := webhook
+
+	if webhook == "test" { // temporary for testing
+		url = "https://hooks.slack.com/services/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"
+	}
+	// os.Getenv("SLACK_URL")
+
+	n := &SlackNotifier{
+		hookURL: url,
+	}
+
+	return n
+}
+
+// Color is used to represent message color for different states (i.e success, fail)
+type Color string
+
+const (
+	// Green when build successful
+	Green Color = "good"
+	// Yellow ...
+	Yellow Color = "warning"
+	// Red when build unsuccessful
+	Red Color = "danger"
+)
+
+// NotifyOptions is used to configure formatting of notifications
+type NotifyOptions struct {
+	Color   Color
+	Warning bool
+}
+
+// MessageArray builds the json message to be posted to webhook
+type MessageArray struct {
+	Attachments []Message `json:"attachments"`
+}
+
+// Message builds the attachments content of Message
+type Message struct {
+	Text  string `json:"text"`
+	Color string `json:"color"`
+}
+
+// Notify sends the notification
+func (n *SlackNotifier) Notify(text string, options *NotifyOptions) error {
+	// check if url is empty
```

Comment:
only really need to leave comments if the code is not self-explanatory 😋 

```suggestion
	// check if url is empty
```